src/Avalonia.Base/Visual.cs

```diff
@@ -655,7 +658,10 @@ private static void ZIndexChanged(AvaloniaPropertyChangedEventArgs e)
         /// <param name="e">The event args.</param>
         private void RenderTransformChanged(object? sender, EventArgs e)
         {
-            InvalidateVisual();
+            if (ClipToBounds && IsEffectivelyVisible)
```

Comment:
ClipToBounds?
IsEffectivelyVisible essentially inherits IsVisible property from all parents, combined. It's not related to the clipping or fact of control being displayed on the screen right or not.